iceoryx2-bb/cxx/include/iox2/container/optional.hpp

```diff
@@ -331,6 +330,20 @@ class Optional {
     constexpr auto reset() noexcept -> void {
         m_value.reset();
     }
+
+  private:
+    friend auto operator==(const Optional<T>& lhs, NulloptT /* unused */) noexcept -> bool {
```

Comment:
This looks so wrong. Is this a free function defined in a templatized class?

Can we please not do this! The first thing I thought was, is this a friend method? What does this mean? And why only for `NulloptT` and not for the other comparision functions.